### PR TITLE
Add javadoc code examples

### LINKED ISSUE #11159: Add expression code examples to the javadoc
Capturing from an user trying out expressions that the API is hard to discover. Adding examples to the javadoc would make the a lot easier. These examples should be added to the data driven property itself. eg. `PropertyFactory#textField`.

### PR DESCRIPTION
**Pull request by tobrun**
WIP, this PR adds inline expression code examples to the javadoc.

closes #11159

### CODE REVIEW

**Comment by tobrun** [comment]
Ready for review @LukasPaczos

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java [hunk 1]
```diff
@@ -1436,6 +2510,17 @@ public static Expression toNumber(@NonNull Expression input) {
   /**
    * "Converts the input value to a boolean. The result is `false` when then input is an empty string, 0, false,
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: Unnecessary quotation mark.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java [hunk 2]
```diff
@@ -1590,6 +2777,22 @@ public static Expression step(@NonNull Expression input, @NonNull Expression def
...
    * The output type must be `number`, `array&lt;number&gt;`, or `color`.
+   * <p>
+   * Example usage:
+   * </p>
+   * <pre>
+   * {@code
+   * FillLayer fillLayer = new fillLayer("layer-id", "source-id");
+   * fillLayer.setProperties(
+   *     fillColor(
+   *         exponential(0.5f), zoom(),
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: Missing `interpolate()` call or is that not mandatory?

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java [hunk 3]
```diff
@@ -1607,6 +2810,22 @@ public static Expression interpolate(@NonNull Interpolator interpolation,
...
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: Same as above.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java [hunk 4]
```diff
@@ -1621,6 +2840,22 @@ public static Expression interpolate(@NonNull Interpolator interpolation,
...
    * interpolates linearly between the pair of stops just less than and just greater than the input.
+   * <p>
+   * Example usage:
+   * </p>
+   * <pre>
+   * {@code
+   * FillLayer fillLayer = new fillLayer("layer-id", "source-id");
+   * fillLayer.setProperties(
+   *     fillColor(
+   *         linear(), zoom(),
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: Same as above.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java [hunk 5]
```diff
@@ -1634,6 +2869,22 @@ public static Interpolator linear() {
...
    * With values close to 1 the output increases linearly.
+   * <p>
+   * Example usage:
+   * </p>
+   * <pre>
+   * {@code
+   * FillLayer fillLayer = new fillLayer("layer-id", "source-id");
+   * fillLayer.setProperties(
+   *     fillColor(
+   *         exponential(0.5f), zoom(),
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: Same as above.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java [hunk 6]
```diff
@@ -1659,6 +2910,22 @@ public static Interpolator exponential(@NonNull Expression expression) {
...
    * Interpolates using the cubic bezier curve defined by the given control points.
+   * <p>
+   * Example usage:
+   * </p>
+   * <pre>
+   * {@code
+   * FillLayer fillLayer = new fillLayer("layer-id", "source-id");
+   * fillLayer.setProperties(
+   *     fillColor(
+   *         cubicBezier(0.42f, 0.0f, 1.0f, 1.0f), zoom(),
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: Same as above.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java [hunk 7]
```diff
@@ -1674,6 +2941,22 @@ public static Interpolator cubicBezier(@NonNull Expression x1, @NonNull Expressi
...
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: Same as above.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java [hunk 8]
```diff
@@ -527,6 +894,22 @@ public static Expression not(boolean input) {
...
+   * <p>
+   * Example usage:
+   * </p>
+   * <pre>
+   * {@code
+   * SymbolLayer symbolLayer = new SymbolLayer("layer-id", "source-id");
+   * symbolLayer.setProperties(
+   *     iconSize(
+   *         switchCase(
+   *             get(KEY_PROPERTY_SELECTED), literal(3.0f),
```
[the remark this answers is not in the document]
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: that would make it a bit more clear + also added `// default value`:
```
   * symbolLayer.setProperties(
   *     iconSize(
   *         switchCase(
   *             get(KEY_TO_BOOLEAN), literal(3.0f),
   *             get(KEY_TO_OTHER_BOOLEAN), literal(5.0f)
   *             literal(1.0f) // default value
   *         )
   *     )
   * );
```

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java [hunk 9]
```diff
@@ -701,6 +1192,17 @@ public static Expression get(@NonNull String key, @NonNull Expression object) {
...
   /**
    * Tests for the presence of an property value in the current feature's properties.
+   * <p>
+   * Example usage:
+   * </p>
+   * <pre>
+   * {@code
+   * FillLayer fillLayer = new FillLayer("layer-id", "source-id");
+   * fillLayer.setFilter(
+   *     has(string("keyToValue"))
```
[the remark this answers is not in the document]
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: actually `has(get("keyToValue"))`

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (LukasPaczos) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java]: Looks great :rocket:

**Pull request merged**